enhancements/network/on-prem-bgp-networking.md

```diff
@@ -0,0 +1,455 @@
+done
+```
+
+##### Health check
+
+When checking for the health of a service (API and Ingress), we'll run a `curl` against the local service (API or Ingress) and check the response code.
+
+As soon as a service becomes unavailable, the VIPs of the service are immediately removed and thanks to BFD, the route to reach the VIPs will be removed on
+the upstream routers. This ensures the high availability of the VIPs.
+
+##### Creating VIPs
+
+VIPs will be created on a loopback device (lo) which is the most popular way because it's simple and we don't need to create a dummy device that consumes memory and reduces the cost of operations and maintenance. We use `ip address` command that is widely available, however we need to find a container image with the `iproute2` package
+or maybe use `baremetal-runtimecfg`.
+
+##### VIPs routing
+
+Since the VIP won't be in the same subnet of the bootstrap and master nodes, we need to configure
+routing on the host as well. We spent a bit of time investigating it and the problem with the VIP
+traffic going back to its source (specially for HTTPS) is that the traffic path is asymmetric.
+The destination MAC on the request was different than the source MAC the client was seeing on the reply;
+and then packets were dropped.
+
+We think that we have two options:
+- Use a specific routing table that will route outgoing traffic coming from the VIP and supposed
+  to go to the nodes subnet, to go through the gateway so that the traffic back follows the same
+  path as the traffic in.
+- For all the hosts in the nodes subnet, provide a direct route to reach the VIP.
+
+We decided to go with the first option (VIP traffic goes through the router in and out) because the VIP will be
+active/active on multiple nodes (masters).
+While this sounds complex, this comes due to abusing the deployment having different subnet IPs
+on the same segment and expect it to not go through the router, so we need to make symmetric traffic.
+
+This implies to create an IP table on each node where FRR will be running, that will enforce the describe behavior:
+```bash
+ip rule add from $VIP to $NETWORK_CIDR lookup <number value>
+ip route add default via $GATEWAY table <number value>
+```
+
+
+#### frr-monitor
+
+This container will have in charge:
+- Generate `frr.conf` configuration file so FRR can work correctly
+- Watch for configuration changes or events that require FRR daemons to stop or reload.
+
+The code will live in baremetal-runtimecfg among other monitors (HAproxy, Keepalived, etc).
+
+### Load balanced VIPs
+
+With Keepalived, the VIPs were located on one (elected) host (using [VRRP](https://fr.wikipedia.org/wiki/Virtual_Router_Redundancy_Protocol)) and HAproxy on that node
+would distribute the traffic across available control plane nodes.
+
+With BGP anycast, the VIPs will reside on all control plane hosts, and upstream routers (e.g. ToR) will define the best route (according to BGP protocol) and depending on how the routers
```

Comment:
So in other words, we'll provide recommendation in the docs on how to configure load-balancing, but ultimately the responsibility falls to the infra admin.

Reply:
I added a sentence explaining that.